camel/messages/base.py

```diff
@@ -26,370 +26,306 @@
-        else:
-            raise TypeError(
-                f"Unsupported operand type(s) for +: '{type(self)}' and "
-                f"'{type(other)}'"
+            new_content = Content(
+                text=(self.content.text or []) + (other.content.text or []),
+                image_url=(self.content.image_url or [])
+                + (other.content.image_url or []),
+                video_url=(self.content.video_url or [])
+                + (other.content.video_url or []),
+                audio_url=(self.content.audio_url or [])
+                + (other.content.audio_url or []),
             )
-        return self.create_new_instance(combined_content)
+            return self.create_new_instance(new_content)
+        raise TypeError(
+            f"Unsupported operand type(s) for"
+            f" +: '{type(self)}' and '{type(other)}'"
+        )
 
     def __mul__(self, other: Any) -> Union["BaseMessage", Any]:
-        r"""Multiplication operator override for :obj:`BaseMessage`.
-
-        Args:
-            other (Any): The value to be multiplied with.
-
-        Returns:
-            Union[BaseMessage, Any]: The result of the multiplication.
-        """
         if isinstance(other, int):
-            multiplied_content = self.content.__mul__(other)
-            return self.create_new_instance(multiplied_content)
-        else:
-            raise TypeError(
-                f"Unsupported operand type(s) for *: '{type(self)}' and "
-                f"'{type(other)}'"
+            new_content = Content(
+                text=(self.content.text or []) * other,
+                image_url=(self.content.image_url or []) * other,
+                video_url=(self.content.video_url or []) * other,
+                audio_url=(self.content.audio_url or []) * other,
             )
+            return self.create_new_instance(new_content)
+        raise TypeError(
+            f"Unsupported operand type(s) for *:"
+            f" '{type(self)}' and '{type(other)}'"
+        )
 
     def __len__(self) -> int:
-        r"""Length operator override for :obj:`BaseMessage`.
-
-        Returns:
-            int: The length of the content.
-        """
-        return len(self.content)
+        return sum(
+            len(getattr(self.content, attr) or [])
+            for attr in ["text", "image_url", "video_url", "audio_url"]
+        )
 
     def __contains__(self, item: str) -> bool:
-        r"""Contains operator override for :obj:`BaseMessage`.
-
-        Args:
-            item (str): The item to check for containment.
-
-        Returns:
-            bool: :obj:`True` if the item is contained in the content,
-                :obj:`False` otherwise.
-        """
-        return item in self.content
+        return any(
+            item in getattr(self.content, attr) or []
+            for attr in ["text", "image_url", "video_url", "audio_url"]
+        )
 
     def extract_text_and_code_prompts(
```

Comment:
I think that it is fine to maintain the functionality since we don't want to break any existing usage of this method.

However if there is a use case, I think that we may need another method since we are supporting multi modal and we will support the structured outputs, which are not limited to text and code.